torch_em/data/image_collection_dataset.py

```diff
@@ -102,17 +102,23 @@ def _get_sample(self, index):
 
         shape = raw.shape
         # we assume images are loaded with channel last!
+        prefix_box = tuple()
         if have_raw_channels:
-            shape = shape[:-1]
+            if shape[-1] < 16:
```

Comment:
I think the heuristic makes sense and implementing proper metadata parsing will be too challenging.
But we should add a comment about it since it may not be quite obvious from the code alone (see my suggestion)

Reply:
It is not so hard for most popular metadata, but there are mant tiff files that do not contain metadata (for example output of `tifffile.imwrite([path, data)` does not contain metadata. One need to specify them as a separate entry.